src/test/java/uk/nhs/cdss/transform/out/CDSOutputTransformerTest.java

```diff
@@ -1,113 +1,349 @@
+    transformedCarePlan1.setId("CarePlan/validCarePlan1");
+    var transformedCarePlan2 = new CareConnectCarePlan();
+    transformedCarePlan2.setId("CarePlan/validCarePlan2");
+
+    var transformedRequestGroup = new RequestGroup();
+    transformedRequestGroup.setId("RequestGroup/validRequestGroup");
+
+    var transformedReferralRequest = new org.hl7.fhir.dstu3.model.ReferralRequest();
+    transformedReferralRequest.setId("ReferralRequest/validReferralRequest");
+
+    when(clock.millis()).thenReturn(FIXED_INSTANT);
+    when(observationTransformer.transform(argThat(isObservationBundleFor(observation1))))
+        .thenReturn(transformedObservation1);
+    when(observationTransformer.transform(argThat(isObservationBundleFor(observation2))))
+        .thenReturn(transformedObservation2);
+    when(storageService.create(argThat(isParametersContaining(
+        isParameter("outputData", transformedObservation1),
+        isParameter("outputData", transformedObservation2),
+        isParameter("outputData", questionnaireResponse1),
+        isParameter("outputData", questionnaireResponse2)
+    )))).thenReturn(new Reference("Parameters/validOutputParameters"));
+    when(requestGroupTransformer.transform(bundle)).thenReturn(transformedRequestGroup);
+    when(carePlanTransformer.transform(argThat(isCarePlanBundleFor(carePlan1, false))))
+        .thenReturn(transformedCarePlan1);
+    when(carePlanTransformer.transform(argThat(isCarePlanBundleFor(carePlan2, false))))
+        .thenReturn(transformedCarePlan2);
+    when(referralRequestTransformer.transform(argThat(isReferralRequestBundleFor(referralRequest, false))))
+        .thenReturn(transformedReferralRequest);
+    when(storageService.create(argThat(sameInstance(transformedCarePlan1))))
+        .thenReturn(new Reference("CarePlan/validCarePlan1"));
+    when(storageService.create(argThat(sameInstance(transformedCarePlan2))))
+        .thenReturn(new Reference("CarePlan/validCarePlan2"));
+    when(storageService.create(argThat(sameInstance(transformedReferralRequest))))
+        .thenReturn(new Reference("ReferralRequest/validReferralRequest"));
+
+    GuidanceResponse guidanceResponse = outputTransformer.transform(bundle);
+
+    assertThat(guidanceResponse.getStatus(), is(GuidanceResponseStatus.SUCCESS));
+    assertThat(guidanceResponse.getOccurrenceDateTime(), sameInstant(FIXED_INSTANT));
+    assertThat(guidanceResponse.getRequestId(), is("validRequestId"));
+    assertThat(guidanceResponse.getModule(), referenceTo("ServiceDefinition/1"));
+    assertThat(guidanceResponse.getContext(), is(context));
+    assertThat(guidanceResponse.getSubject(), is(patient));
+    assertThat(guidanceResponse.getOutputParameters(), referenceTo("Parameters/validOutputParameters"));
+    assertThat(guidanceResponse.getResult(), referenceTo("RequestGroup/validRequestGroup"));
+
+    verify(storageService).upsert(transformedRequestGroup);
+    assertThat(transformedRequestGroup.getAction(), hasItems(
```

Comment:
using the `contains` matcher would be more appropriate. `hasItems` seems to check that the items appear a minimum of once, rather than exactly the amount of times we provide it.